src/main.rs

```diff
@@ -93,46 +102,299 @@ macro_rules! create_socket {
+
+        let gpioa = dp.GPIOA.split(&mut clocks.ahb4);
+        let gpiob = dp.GPIOB.split(&mut clocks.ahb4);
+        let gpioc = dp.GPIOC.split(&mut clocks.ahb4);
+        let gpiod = dp.GPIOD.split(&mut clocks.ahb4);
+        let gpioe = dp.GPIOE.split(&mut clocks.ahb4);
+        let gpiof = dp.GPIOF.split(&mut clocks.ahb4);
+        let gpiog = dp.GPIOG.split(&mut clocks.ahb4);
+
+        // Configure the SPI interfaces to the ADCs and DACs.
+        let adc1_spi = {
+            let spi_miso = gpiob.pb14.into_alternate_af5().set_speed(hal::gpio::Speed::VeryHigh);
+            let spi_sck = gpiob.pb10.into_alternate_af5().set_speed(hal::gpio::Speed::VeryHigh);
+            let _spi_nss = gpiob.pb9.into_alternate_af5();
+
+            let config = hal::spi::Config::new(hal::spi::Mode{
+                    polarity: hal::spi::Polarity::IdleHigh,
+                    phase: hal::spi::Phase::CaptureOnSecondTransition,
+                })
+                .communication_mode(hal::spi::CommunicationMode::Receiver)
+                .manage_cs()
+                .cs_delay(220e-9)
+                .frame_size(16);
+
+            let mut spi = dp.SPI2.spi(
+                    (spi_sck, spi_miso, hal::spi::NoMosi),
+                    config,
+                    50.mhz(),
+                    &clocks);
+
+            spi.listen(hal::spi::Event::Rxp);
+
+            spi
+        };
+
+        let adc2_spi = {
+            let spi_miso = gpiob.pb4.into_alternate_af6().set_speed(hal::gpio::Speed::VeryHigh);
+            let spi_sck = gpioc.pc10.into_alternate_af6().set_speed(hal::gpio::Speed::VeryHigh);
+            let _spi_nss = gpioa.pa15.into_alternate_af6();
+
+
+            let config = hal::spi::Config::new(hal::spi::Mode{
+                    polarity: hal::spi::Polarity::IdleHigh,
+                    phase: hal::spi::Phase::CaptureOnSecondTransition,
+                })
+                .communication_mode(hal::spi::CommunicationMode::Receiver)
+                .manage_cs()
+                .frame_size(16)
+                .cs_delay(220e-9);
+
+            let spi = dp.SPI3.spi(
+                    (spi_sck, spi_miso, hal::spi::NoMosi),
+                    config,
+                    50.mhz(),
+                    &clocks);
+
+            spi
+        };
+
+        let dac1_spi = {
+            let spi_miso = gpioe.pe5.into_alternate_af5();
+            let spi_sck = gpioe.pe2.into_alternate_af5();
+            let _spi_nss = gpioe.pe4.into_alternate_af5();
+
+            let config = hal::spi::Config::new(hal::spi::Mode{
+                    polarity: hal::spi::Polarity::IdleHigh,
+                    phase: hal::spi::Phase::CaptureOnSecondTransition,
+                })
+                .communication_mode(hal::spi::CommunicationMode::Transmitter)
+                .manage_cs()
+                .frame_size(16)
+                .swap_mosi_miso();
+
+            dp.SPI4.spi((spi_sck, spi_miso, hal::spi::NoMosi), config, 25.mhz(), &clocks)
```

Comment:
The DACs can and should do 50 MHz (div2). As in the old firmware.
https://www.analog.com/media/en/technical-documentation/data-sheets/AD5512A_5542A.pdf